libmediation/types/types.cpp

```diff
@@ -350,11 +361,22 @@ string closeDirPath(const string& src, char delimiter)
         delimiter = getDirSeparator();
     if (src.length() == 0)
         return src;
-    if (src[src.length() - 1] == '/' || src[src.length() - 1] == '\\')
+    if (src[src.length() - 1] == getDirSeparator())
         return src;
     return src + delimiter;
 }
 
+// extract the filename from a path, check for invalid characters
+bool isValidFileName(const string& src)
+{
+    string filename = extractFileName(src);
+
+    // invalidChars() returns a different regex pattern for Windows or Unix
+    regex ourInvalidChars = invalidChars();
```

Comment:
The type will need to be changed to `auto&&` or `const regex&` if you choose to follow my suggestion about the return value of `invalidChars()`. Or just directly call `invalidChars()` in the `regex_search` call and avoid the temporary variable altogether.